x/mongo/driver/topology/server.go

```diff
@@ -295,11 +318,25 @@ func (s *Server) RequestImmediateCheck() {
 }
 
 // ProcessError handles SDAM error handling and implements driver.ErrorProcessor.
-func (s *Server) ProcessError(err error) {
-	desc := s.Description()
+func (s *Server) ProcessError(err error, conn driver.Connection) {
+	s.processErrorLock.Lock()
+	defer s.processErrorLock.Unlock()
+
+	staleConnection := conn.Stale()
+
 	// Invalidate server description if not master or node recovering error occurs.
 	// These errors can be reported as a command error or a write concern error.
 	if cerr, ok := err.(driver.Error); ok && (cerr.NodeIsRecovering() || cerr.NotMaster()) {
+		desc := s.Description()
+
+		// ignore stale error
+		if staleConnection || desc.TopologyVersion.MoreRecentThan(cerr.TopologyVersion) {
```

Comment:
Equal topologyVersions should be considered stale here. Is ``desc.TopologyVersion.MoreRecentThan(cerr.TopologyVersion)``  true when ``desc.TopologyVersion== cerr.TopologyVersion``?

Reply:
This is kind of annoying, but this code is actually from the "Reduce race conditions in SDAM error handling" ticket. There's a PR for that ticket at https://github.com/mongodb/mongo-go-driver/pull/425 and the corresponding code in that PR actually uses a `CompareTopologyVersions` function to return (-1, 0, 1). This PR is based off of an older version of that ticket because I needed something to get the integration tests passing.